### PR TITLE
Bug fix/handle setup test errors

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

if test setup fails fataly by throwing we should make it fail

- [x] :hankey: Bugfix

### CODE REVIEW

**Review by jsteemann — APPROVED** [review]
LGTM

File: tests/js/server/recovery/view-search-alias-populate-transaction-no-flushthread.js [hunk 1]
```diff
@@ -54,6 +54,7 @@ function runSetup () {
       write: ['UnitTestsRecoveryDummy']
     },
     action: function() {
+      var db = require('internal').db;
```
Comment (MBkkt) [inline tests/js/server/recovery/view-search-alias-populate-transaction-no-flushthread.js:57]: NIT
```suggestion
      const db = require('internal').db;
```

File: tests/js/server/recovery/view-search-alias-populate-no-sync.js [hunk 2]
```diff
@@ -48,11 +48,31 @@ function runSetup () {
...
+    }
+  }
+
+  try {
+    c.save({ name: 'crashme' }, { waitForSync: true });
+  } catch (ex) {
+    if (ex.errorNum !== internal.errors.ERROR_SIMPLE_CLIENT_UNKNOWN_ERROR.code) {
+      print(ex);
+      throw ex;
+    }
```
Comment (MBkkt) [inline tests/js/server/recovery/view-search-alias-populate-no-sync.js:73]:
```suggestion
  try {
    for (let i = 0; i < 50000; i++) {
      c.save({ a: "foo_" + i, b: "bar_" + i, c: i });
    }
    c.save({ name: 'crashme' }, { waitForSync: true });
  } catch (ex) {
    if (ex.errorNum !== internal.errors.ERROR_SIMPLE_CLIENT_UNKNOWN_ERROR.code) {
      print(ex);
      throw ex;
    }
```

File: tests/js/server/recovery/view-search-alias-populate-no-sync.js [hunk 3]
```diff
@@ -48,11 +48,31 @@ function runSetup () {
   internal.wal.flush(true, true);
   internal.debugSetFailAt("FlushCrashBeforeSyncingMinTick");
 
-  for (let i = 0; i < 50000; i++) {
-    c.save({ a: "foo_" + i, b: "bar_" + i, c: i });
+  if (global.hasOwnProperty('arango')) {
+    // we intend to crash, so we should get to know quickly:
+    print('setting short timeout');
+    arango.timeout(1);
+  }
```
Comment (MBkkt) [inline tests/js/server/recovery/view-search-alias-populate-no-sync.js:55]: hmm, but what if crash not really happened and we just faced with slow requests in our CI?

Maybe remove it?
It's just optimization

File: tests/js/server/recovery/view-search-alias-populate-no-release.js [hunk 4]
```diff
@@ -48,11 +48,30 @@ function runSetup () {
...
 
-  c.save({ name: 'crashme' }, { waitForSync: true });
+  try {
+    c.save({ name: 'crashme' }, { waitForSync: true });
+  } catch (ex) {
+    if (ex.errorNum !== internal.errors.ERROR_SIMPLE_CLIENT_UNKNOWN_ERROR.code) {
+      print(ex);
+      throw ex;
+    }
+  }
```
Comment (MBkkt) [inline tests/js/server/recovery/view-search-alias-populate-no-release.js:74]: unify this with code above

File: tests/js/server/recovery/view-search-alias-populate-no-callbacks.js [hunk 5]
```diff
@@ -48,12 +48,30 @@ function runSetup () {
   internal.wal.flush(true, true);
   internal.debugSetFailAt("FlushCrashBeforeSyncingMinTick");
 
-  for (let i = 0; i < 100000; i++) {
-    c.save({ a: "foo_" + i, b: "bar_" + i, c: i });
+  if (global.hasOwnProperty('arango')) {
+    // we intend to crash, so we should get to know quickly:
+    print('setting short timeout');
+    arango.timeout(1);
+  }
```
Comment (MBkkt) [inline tests/js/server/recovery/view-search-alias-populate-no-callbacks.js:55]: same

**Review by MBkkt — COMMENTED** [review]
For all tests are same issues.
Also maybe you can make single function for this try catch logic?

**Review by MBkkt — CHANGES_REQUESTED** [review]
Because Jan already approved, but I think we need to make it more clear and probably without timeout change

File: tests/js/server/recovery/view-search-alias-populate-no-sync.js (hunk 3 above)
Comment (dothebart) [reply tests/js/server/recovery/view-search-alias-populate-no-sync.js:55]: I'd say 1s still is plenty, right?

**Review by MBkkt — APPROVED** [review]

**Comment by KVS85** [comment]
- [ ] https://jenkins01.arangodb.biz/job/arangodb-ANY-jslint/28768/

**Pull request merged**